dynawo/sources/Modeler/Common/test/Test.cpp

```diff
@@ -1106,4 +1117,33 @@ TEST(ModelerCommonTest, StaticRefInterface) {
   ASSERT_EQ(sri.getModelVar(), "MyModelVar");
   ASSERT_EQ(sri.getStaticVar(), "MyStaticVar");
 }
+
+TEST(ModelerCommonTest, ConnectionCalculatedVars) {
+  boost::shared_ptr<ModelMulti> modelMulti(new ModelMulti());
+  shared_ptr<SubModelMock> subModel1(new SubModelMock(1, 1));
+  subModel1->name("subModel1");
+  shared_ptr<SubModel> subModel1_ = boost::dynamic_pointer_cast<SubModel> (subModel1);
+  subModel1_->name("subModel1");
+  subModel1_->defineNames();
+  subModel1_->defineVariables();
+  modelMulti->addSubModel(subModel1_, "subModelMock");
+
+  shared_ptr<SubModelMock> subModel2(new SubModelMock(1, 1));
+  subModel2->name("subModel2");
+  shared_ptr<SubModel> subModel2_ = boost::dynamic_pointer_cast<SubModel> (subModel2);
+  subModel2_->name("subModel2");
+  subModel2_->defineNames();
+  subModel2_->defineVariables();
+  modelMulti->addSubModel(subModel2_, "subModelMock");
+
+  std::vector<double> y, yp, z;
+  modelMulti->copyContinuousVariables(&y[0], &yp[0]);
+  modelMulti->copyDiscreteVariables(&z[0]);
+  ASSERT_NO_THROW(modelMulti->connectElements(modelMulti->findSubModelByName("subModel1"), "MyDiscreteVarCalculated_value",
+                              modelMulti->findSubModelByName("subModel2"), "MyDiscreteVar_value"));
```

Comment:
Would it be possible to write some unit tests about the new submodel? (testing the root is raised when required ...)